src/PrestaShopBundle/Resources/config/services/core/cldr.yml

```diff
@@ -1,8 +1,101 @@
-services:
-    _defaults:
-        public: true
+parameters:
+    ps_cache_dir: !php/const _PS_CACHE_DIR_
 
+services:
     prestashop.core.cldr.repository:
         class: 'PrestaShop\PrestaShop\Core\Cldr\Repository'
         arguments:
-            - '@=service("prestashop.adapter.legacy.context").getLanguage()'
+          - '@=service("prestashop.adapter.legacy.context").getLanguage()'
+    # Data providers service
+    prestashop.adapter.data_provider.currency:
+        class: PrestaShop\PrestaShop\Adapter\Currency\CurrencyDataProvider
+        arguments:
+          - "@prestashop.adapter.legacy.configuration"
+          - "@=service('prestashop.adapter.legacy.context').getContext().shop?service('prestashop.adapter.legacy.context').getContext().shop.id"
+
+    prestashop.adapter.legacy.configuration:
```

Comment:
adapter services shouldn't be in core config
besides this creates duplicates as these services are already defined in `src/PrestaShopBundle/Resources/config/services/adapter/services.yml`

Reply:
duplicates for `prestashop.adapter.legacy.context` and `prestashop.adapter.legacy.configuration`
but `prestashop.adapter.data_provider.currency` should be defined in an adapter config file